workflow/controller/cache_gc.go

```diff
@@ -0,0 +1,72 @@
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/argoproj/argo-workflows/v3/util/env"
+	controllercache "github.com/argoproj/argo-workflows/v3/workflow/controller/cache"
+	"github.com/argoproj/argo-workflows/v3/workflow/controller/indexes"
+)
+
+// SyncAllCacheForGC syncs all cache for GC
+func SyncAllCacheForGC(ctx context.Context, configMapInformer cache.SharedIndexInformer, kubeclientset kubernetes.Interface) {
```

Comment:
what not main this a func of `WorkflowController`?  that way you have all the informers etc to hand with `wfc.*`